chainer/functions/array/stack.py

```diff
@@ -6,12 +6,61 @@ def stack(xs, axis=0):
     """Concatenate variables along a new axis.
 
     Args:
-        xs (list of chainer.Variable): Variables to be concatenated.
-        axis (int): Axis of result along which variables are stacked.
+        xs (list of :class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`):
+            Input variables to be concatenated. The variables must have the
+            same shape.
+        axis (int): The axis along which the arrays will be stacked. Default
+            is 0.
```

Comment:
Please write the restriction of the value of `axis`. At least, `0<= axis < ndim` is acceptable and `axis >= ndim` is not allowed. I did not investigated negative axis yet.